zap/src/main/java/org/zaproxy/zap/authentication/AuthenticationMethod.java

```diff
@@ -42,16 +51,61 @@
  */
 public abstract class AuthenticationMethod {
 
+    private static final Logger LOGGER = Logger.getLogger(AuthenticationMethod.class);
+
     public static final String CONTEXT_CONFIG_AUTH = Context.CONTEXT_CONFIG + ".authentication";
     public static final String CONTEXT_CONFIG_AUTH_TYPE = CONTEXT_CONFIG_AUTH + ".type";
+    public static final String CONTEXT_CONFIG_AUTH_STRATEGY = CONTEXT_CONFIG_AUTH + ".strategy";
+    public static final String CONTEXT_CONFIG_AUTH_POLL_URL = CONTEXT_CONFIG_AUTH + ".pollurl";
+    public static final String CONTEXT_CONFIG_AUTH_POLL_DATA = CONTEXT_CONFIG_AUTH + ".polldata";
+    public static final String CONTEXT_CONFIG_AUTH_POLL_HEADERS =
+            CONTEXT_CONFIG_AUTH + ".pollheaders";
+    public static final String CONTEXT_CONFIG_AUTH_POLL_FREQ = CONTEXT_CONFIG_AUTH + ".pollfreq";
+    public static final String CONTEXT_CONFIG_AUTH_POLL_UNITS = CONTEXT_CONFIG_AUTH + ".pollunits";
     public static final String CONTEXT_CONFIG_AUTH_LOGGEDIN = CONTEXT_CONFIG_AUTH + ".loggedin";
     public static final String CONTEXT_CONFIG_AUTH_LOGGEDOUT = CONTEXT_CONFIG_AUTH + ".loggedout";
 
+    public static final String AUTH_STATE_ASSUMED_IN_STATS = "stats.auth.state.assumedin";
     public static final String AUTH_STATE_LOGGED_IN_STATS = "stats.auth.state.loggedin";
     public static final String AUTH_STATE_LOGGED_OUT_STATS = "stats.auth.state.loggedout";
     public static final String AUTH_STATE_NO_INDICATOR_STATS = "stats.auth.state.noindicator";
     public static final String AUTH_STATE_UNKNOWN_STATS = "stats.auth.state.unknown";
 
+    public static final String TOKEN_PREFIX = "{%";
+    public static final String TOKEN_POSTFIX = "%}";
+
+    public static final int DEFAULT_POLL_FREQUENCY = 60;
+
+    public static enum AuthCheckingStrategy {
+        EACH_RESP,
+        EACH_REQ,
+        EACH_REQ_RESP,
+        POLL_URL
+    };
+
+    public static enum AuthPollFrequencyUnits {
+        REQUESTS,
+        SECONDS
+    };
+
+    private AuthCheckingStrategy authCheckingStrategy = AuthCheckingStrategy.EACH_RESP;
+
+    private String pollUrl;
+
+    private String pollData;
+
+    private String pollHeaders;
+
+    private int pollFrequency = DEFAULT_POLL_FREQUENCY;
+
+    private AuthPollFrequencyUnits pollFrequencyUnits = AuthPollFrequencyUnits.REQUESTS;
+
+    private Date lastPollTime;
```

Comment:
Could be a `long` and use `System.currentTimeMillis()`, the `Date` is not actually being used just its time.